std/repl.glu

```diff
@@ -94,14 +94,14 @@ let loop x: () -> IO () =
                 >>= io.print
                 >> return True
 
-    repl_prim.input "> " >>= \line_opt ->
+    rustyline.readline editor "> " >>= \line_opt ->
         match line_opt with
             | None -> return ()
             | Some line -> run_line line >>= \continue ->
-                if continue then loop () else return ()
+                if continue then loop editor else return ()
 
 let run x: () -> IO () =
     io.print "gluon (:h for help, :q to quit)"
-        >>= loop
+        >>= \_-> loop (rustyline.new_editor ())
```

Comment:
I did a pretty through explanation of my reasoning in a hackenews thread. Let me know if you have any thoughts or questions as it's not set in stone. https://news.ycombinator.com/item?id=12088664